pkg/services/object/get/assembly_v2.go

```diff
@@ -0,0 +1,178 @@
+package getsvc
+
+import (
+	"errors"
+
+	apistatus "github.com/nspcc-dev/neofs-sdk-go/client/status"
+	objectSDK "github.com/nspcc-dev/neofs-sdk-go/object"
+	oid "github.com/nspcc-dev/neofs-sdk-go/object/id"
+	"go.uber.org/zap"
+)
+
+func (exec *execCtx) processV2Split(si *objectSDK.SplitInfo) {
+	if _, set := si.FirstPart(); !set {
+		exec.log.Debug("no first ID found in V2 split")
+		exec.err = errors.New("v2 split without first object ID")
+
+		return
+	}
+
+	linkID, set := si.Link()
+	if set && exec.processV2Link(linkID) {
+		return
+	}
+
+	// fallback to the full chain assembly from the last part
+	prev, set := si.LastPart()
+	if set {
```

Comment:
else?

Reply:
just `return`. that is how `execCtx` works. if no status and result is set inside some of the func it returns 404. this line is the last. but agree, added Debug logs here